libs/dml/src/native_type_instance.rs

```diff
@@ -1,34 +1,44 @@
-use native_types::NativeType;
-use serde::de::DeserializeOwned;
+use psl_core::datamodel_connector::{Connector, NativeTypeInstance as PslNativeTypeInstance};
+use std::fmt;
 
-/// represents an instance of a native type declared in the Prisma schema
-#[derive(Debug, Clone, PartialEq)]
+/// Represents an instance of a native type declared in the Prisma schema.
+#[derive(Clone)]
 pub struct NativeTypeInstance {
-    /// the name of the native type used in the Prisma schema
-    pub name: String,
-    /// the arguments that were provided
-    pub args: Vec<String>,
-    /// the serialized representation of this native type. The serialized format is generated by the `native-types` library
-    pub serialized_native_type: serde_json::Value,
+    pub(crate) native_type: PslNativeTypeInstance,
+    pub(crate) connector: &'static dyn Connector,
+}
+
+impl fmt::Debug for NativeTypeInstance {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(&self.connector.native_type_to_string(&self.native_type))
+    }
+}
+
+impl PartialEq for NativeTypeInstance {
+    fn eq(&self, other: &Self) -> bool {
+        self.connector.native_type_to_parts(&self.native_type)
+            == other.connector.native_type_to_parts(&other.native_type)
+    }
 }
 
 impl NativeTypeInstance {
-    pub fn new(name: &str, args: Vec<String>, native_type: &dyn NativeType) -> Self {
-        NativeTypeInstance {
-            name: name.to_string(),
-            args,
-            serialized_native_type: native_type.to_json(),
-        }
+    pub fn new(native_type: PslNativeTypeInstance, connector: &'static dyn Connector) -> Self {
+        NativeTypeInstance { native_type, connector }
+    }
+
+    pub fn inner(&self) -> &PslNativeTypeInstance {
+        &self.native_type
+    }
+
+    pub fn deserialize_native_type<T: std::any::Any>(&self) -> &T {
```

Comment:
The name isn't accurate anymore, but it avoids more diff. I think we should rename.